plugins/inputs/internet_speed/internet_speed.go

```diff
@@ -79,32 +110,42 @@ func (is *InternetSpeed) Gather(acc telegraf.Accumulator) error {
 	tags := map[string]string{
 		"server_id": is.server.ID,
 		"source":    is.server.Host,
+		"test_mode": is.TestMode,
 	}
-	// recycle the detailed data of each test to prevent data backlog
+	// Recycle the history of each test to prevent data backlog.
 	is.server.Context.Reset()
 	acc.AddFields(measurement, fields, tags)
 	return nil
 }
 
 func (is *InternetSpeed) findClosestServer() error {
-	user, err := speedtest.FetchUserInfo()
+	client := speedtest.New(speedtest.WithUserConfig(&speedtest.UserConfig{
+		UserAgent:  "Telegraf SpeedTestGO/" + speedtest.Version(),
```

Comment:
Why not using ProductToken? that's exactly for this use-case.
```suggestion
		UserAgent:  internal.ProductToken(),
```